src/configure/clients/restClient.ts

```diff
@@ -20,6 +20,21 @@ export class RestClient extends ServiceClient {
         });
     }
 
+    public sendRequest3<TResult>(options: PathTemplateBasedRequestPrepareOptions | UrlBasedRequestPrepareOptions): Promise<TResult> {
```

Comment:
why do we need a new sendrequest implementation?

Reply:
The API call for creating new repo can fail due to multiple reasons, including the existence of repo of the same name. In case of an existing repo name, we need to make API call again with "local repo name + org name" but in case of any other failure, the process needs to be terminated. For deciding the right condition for termination, error code is needed that is why the entire `response` needs to be returned instead of `response.body`.